server/client.go

```diff
@@ -281,8 +281,12 @@ func (c *client) readLoop() {
 		c.cache.subs = 0
 
 		if err := c.parse(b[:n]); err != nil {
-			// handled inline
-			if err != ErrMaxPayload && err != ErrAuthorization {
+			// If client connection has been closed, simply return,
+			// otherwise report a generic parsing error.
+			c.mu.Lock()
+			closed := c.nc == nil
+			c.mu.Unlock()
+			if !closed {
```

Comment:
We had the checks for those two errors for a reason, why do we not need them anymore?